File: CS/MainDemo.Module/BusinessObjects/Department.cs
using System;
using DevExpress.Xpo;
using System.ComponentModel;
using System.Collections.Generic;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace MainDemo.Module.BusinessObjects {
    [DefaultClassOptions]
    [ImageName("BO_Department")]
    [DefaultProperty("Title")]
    public class Department : BaseObject {
        private string title;
        private string office;
        public Department(Session session)
            : base(session) {
        }
        public string Title {
            get {
                return title;
            }
            set {
                SetPropertyValue("Title", ref title, value);
            }
        }
        public string Office {
            get {
                return office;
            }
            set {
                SetPropertyValue("Office", ref office, value);
            }
        }
        [Association]
        public XPCollection<Employee> Employees {
            get {
                return GetCollection<Employee>("Employees");
            }
        }
    }
}

File: VB/MainDemo.Module/BusinessObjects/Department.vb
Imports System
Imports DevExpress.Xpo
Imports System.ComponentModel
Imports System.Collections.Generic
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace MainDemo.[Module].BusinessObjects
    <DefaultClassOptions>
    <ImageName("BO_Department")>
    <DefaultProperty("Title")>
    Public Class Department
        Inherits BaseObject

        Private _title As String
        Private _office As String

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property Title As String
            Get
                Return _title
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Title", _title, value)
            End Set
        End Property

        Public Property Office As String
            Get
                Return _office
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Office", _office, value)
            End Set
        End Property

        <Association>
        Public ReadOnly Property Employees As XPCollection(Of Employee)
            Get
                Return GetCollection(Of Employee)("Employees")
            End Get
        End Property
    End Class
End Namespace
